pandas/core/strings.py

```diff
@@ -1815,36 +1817,34 @@ def __init__(self, data):
     def _validate(data):
         from pandas.core.index import Index
 
-        if (isinstance(data, ABCSeries) and
-                not ((is_categorical_dtype(data.dtype) and
-                      is_object_dtype(data.values.categories)) or
-                     (is_object_dtype(data.dtype)))):
-            # it's neither a string series not a categorical series with
-            # strings inside the categories.
-            # this really should exclude all series with any non-string values
-            # (instead of test for object dtype), but that isn't practical for
-            # performance reasons until we have a str dtype (GH 9343)
+        if isinstance(data, ABCMultiIndex):
```

Comment:
use ``ABCIndex`` (which is distinct from ``ABCIndexClass``, this specifically will not include any other index types as well. Though may need to allow ``CategoricalIndex`` that is inferred to be of the correct type as well.

Reply:
I don't understand - this check is only for `MultiIndex` (the diff wrongly suggests that it replaces the `Series`-branch)